src/isolate.rs

```diff
@@ -1577,6 +1635,16 @@ impl OwnedIsolate {
     std::mem::forget(self);
     snapshot_creator.create_blob(function_code_handling)
   }
+
+  /// Converts this isolate to an isolate that can be shared via the locker api.
+  pub fn to_shared(self) -> SharedIsolate {
+    SharedIsolate::new(self)
+  }
+
+  /// If this isolate is currently locked by the locker api to the current thread.
+  pub fn is_locked(&self) -> bool {
```

Comment:
If someone has access to the isolate in a shared isolate, they are doing that through the locker api already. I'll remove this from `OwnedIsolate`.

I'll add it to `IsolateHandle` instead, that way it's easier to check if an isolate is entered via the thread safe handle.